differ/differ.go

```diff
@@ -159,135 +153,74 @@ func shaCompare(a, b [sha512.Size]byte) bool {
 		} else {
-			hlv2, _ = hlv.NewHLV(bucketUUID2, cas2, 0, "", 0, nil, nil)
+			newHlv, _ := hlv.NewHLV(bucketUUID2, cas2, 0, "", 0, nil, nil)
+			crMeta2.SetHLV(newHlv)
 		}
-	} else if hlv1 == nil && hlv2 != nil { //this is the case where the document is replicated from source to target
+	}
+	if hlv1 == nil && hlv2 != nil { //this is the case where the document is replicated from source to target
 		if hlv2.GetCvSrc() != bucketUUID1 || hlv2.GetCvVer() != cas1 {
-			return false
+			return fmt.Errorf("Implicit construction of source HLV not possible")
 		} else {
-			hlv1, _ = hlv.NewHLV(bucketUUID1, cas1, 0, "", 0, nil, nil)
+			newHlv, _ := hlv.NewHLV(bucketUUID1, cas1, 0, "", 0, nil, nil)
+			crMeta1.SetHLV(newHlv)
 		}
 	}
-	return compareHlvItems(hlv1, hlv2, cas1, cas2)
-}
-
-func compareHlvItems(item1 *hlv.HLV, item2 *hlv.HLV, item1Cas uint64, item2Cas uint64) bool {
-	// the HLVs that this function recieves is up to date
-	if item1.GetCvCas() != item2.GetCvCas() {
-		return false
-	} else if item1.GetCvSrc() != item2.GetCvSrc() || item1.GetCvVer() != item2.GetCvVer() {
-		return false
-	} else if !comparePv(item1.GetPV(), item2.GetPV(), item1Cas, item2Cas) {
-		return false
-	}
-	return true
-}
-
-// Darshan:TODO Accomodate the CAS delta computation inorder to reduce the PV size MB-60961
-func comparePv(Pv1 hlv.VersionsMap, Pv2 hlv.VersionsMap, cas1 uint64, cas2 uint64) bool {
-	if len(Pv1) == len(Pv2) {
-		for key, value1 := range Pv1 {
-			value2, ok := Pv2[key]
-			if !ok {
-				return false
-			} else {
-				if value1 != value2 {
-					return false
-				}
-			}
-		}
-	} else { // Unequal length implies that the PVs are pruned
-		srcPruningWindow := sourcePruningWindow.get()
-		tgtPruningWindow := targetPruningWindow.get()
-		var pruningWindow time.Duration
-		iteratePv := Pv1
-		otherPv := Pv2
-		cas := cas1
-		pruningWindow = srcPruningWindow
-		if len(Pv2) > len(Pv1) {
-			iteratePv = Pv2
-			otherPv = Pv1
-			cas = cas2
-			pruningWindow = tgtPruningWindow
-		}
-		for key, value1 := range iteratePv {
-			value2, ok := otherPv[key]
-			if !ok {
-				if xdcrBase.CasDuration(value1, cas) < pruningWindow {
-					return false
-				}
-				continue
-			}
-			if value1 != value2 {
-				return false
-			}
-		}
-	}
-	return true
+	return nil
 }
 
 // Darshan:TODO Add compare MV when merge is implemented
-// Darshan:TODO Add the prev Revid support when MB-60385 is checked-in
-func (entry oneEntry) GetRevId() uint64 {
-	if entry.ImportCas == entry.Cas {
-		return 0
-	} else {
-		return entry.RevId
-	}
-}
-
-func (entry oneEntry) GetVersion() uint64 {
-	if entry.ImportCas == 0 {
-		return entry.Cas
-	} else {
-		if entry.Cas == entry.ImportCas {
-			return entry.Hlv.GetCvCas()
-		} else if entry.Cas < entry.ImportCas {
-			panic("Import Cas never be greater than the doc cas")
-		} else {
-			return entry.Cas
-		}
+func SetVersion(crMetadata *crMeta.CRMetadata, pRev uint64) {
```

Comment:
This seems out of place or a duplicate work of `UpdateHLVIfNeeded` logic ...

Reply:
I have addressed this in my reply to previous comment